### PR TITLE
adding new functionality for verify checksum

### PR DESCRIPTION
**Pull request by ColeHiggins2**
Adding functionality to Verify Content Checksum

### CODE REVIEW

File: airgun/entities/product.py [hunk 1]
```diff
@@ -123,6 +123,27 @@ def advanced_sync(self, entities_list, sync_type):
...
         return view.read()
 
+    def Verify_content_checksum(self, entities_list):
+        """Advanced Sync for product/products
+
+        :param entities_list: The product names to perform Advanced Sync action.
+        :param sync_type: value containing sync type.
+            eg: sync_type="optimized", sync_type="complete"
+        """
+
```
Comment (latran) [inline airgun/entities/product.py:133]: You're missing the portion where you need to search for the product name and then selecting it.  Here's a sample code to get you started from `common.py` that is used in `contentview.py` in `view`.  Hope this helps!

**Comment by latran** [comment]
So something strange is happening here https://github.com/SatelliteQE/airgun/blob/master/airgun/entities/product.py#L209.  Automation hit this portion, but the selection of the element in the table is not being registered.  Weirdly, if you run `test_positive_bulk_action_advanced_sync` with just `session.product.advanced_sync([product.name], sync_type='optimized')` for https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/ui/test_product.py#L175,  it's able to recognize and select it correctly.  I'm baffled as to why that is.
@omkarkhatavkar can you take a look and see?

**Comment by omkarkhatavkar** [comment]
> So something strange is happening here https://github.com/SatelliteQE/airgun/blob/master/airgun/entities/product.py#L209. Automation hit this portion, but the selection of the element in the table is not being registered. Weirdly, if you run `test_positive_bulk_action_advanced_sync` with just `session.product.advanced_sync([product.name], sync_type='optimized')` for https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/ui/test_product.py#L175, it's able to recognize and select it correctly. I'm baffled as to why that is. @omkarkhatavkar can you take a look and see?

It's not usual behavior from the selenium end. Some time selenium fails to perform actions on the ng (angular) object. Not pretty sure why this happen but I have solution for this here https://github.com/SatelliteQE/airgun/pull/722

File: airgun/entities/product.py [hunk 2]
```diff
@@ -123,6 +123,27 @@ def advanced_sync(self, entities_list, sync_type):
...
+            'Select Action',
+            action_name='Verify Content Checksum',
+            entities_list=entities_list,
+        )
+        #view.select_all.fill(True)
+        view.title.click()
+        view.task.click()
+        view = TaskDetailsView(view.browser)
+        view.wait_for_result()
+        return view.read()
```
Comment (Griffin-Sullivan) [inline airgun/entities/product.py:145]:
```def verify_content_checksum(self, entities_list):
        """Verify content checksum for product
        """

        view = self.navigate_to(
            self,
            'Select Action',
            action_name='Verify Content Checksum',
            entities_list=entities_list,
        )
        view.task_alert.click()
        view = TaskDetailsView(view.browser)
        view.wait_for_result()
        return view.read()
```

Comment (Griffin-Sullivan) [inline airgun/entities/product.py:145]:
```suggestion
    def verify_content_checksum(self, entities_list):
        """Verify content checksum for product
        """

        view = self.navigate_to(
            self,
            'Select Action',
            action_name='Verify Content Checksum',
            entities_list=entities_list,
        )
        view.task_alert.click()
        view = TaskDetailsView(view.browser)
        view.wait_for_result()
        return view.read()
```

File: airgun/views/product.py [hunk 3]
```diff
@@ -261,3 +261,17 @@ class ProductAdvancedSync(BaseLoggedInView):
...
+    """Represents Advanced Sync page for Products."""
+
+    #title = Text("//h4[text()='Verify Content Checksum']")
+    title = Text("//h4[normalize-space(.)='Verify Content Checksum']")
+    task = Text("//a[normalize-space(.)='Click to monitor task progress.']")
+    #sync = Text('//button[@ng-click="ok()"]')
+
+    @property
+    def is_displayed(self):
+        return self.browser.wait_for_element(self.title, exception=False) is not None
```
Comment (Griffin-Sullivan) [inline airgun/views/product.py:277]:
```suggestion
class ProductVerifyContentChecksum(BaseLoggedInView):
    """Represents Advanced Sync page for Products."""

    task_alert = Text("//a[normalize-space(.)='Click to monitor task progress.']")

    @property
    def is_displayed(self):
        return self.browser.wait_for_element(self.title, exception=False) is not None 
```
This is just targeting the button in the alert. I'm guessing someone has already written a view for this that we can steal.

**Review by Griffin-Sullivan — APPROVED** [review]

File: airgun/views/product.py [hunk 4]
```diff
@@ -261,3 +261,13 @@ class ProductAdvancedSync(BaseLoggedInView):
...
+
+
+class ProductVerifyContentChecksum(BaseLoggedInView):
+    """Represents Verify Content Checksum Alert page for Products."""
+
+    task_alert = Text("//a[normalize-space(.)='Click to monitor task progress.']")
+
+    @property
+    def is_displayed(self):
+        return self.browser.wait_for_element(self.title, exception=False) is not None
```
Comment (jyejare) [inline airgun/views/product.py:273]:
```suggestion
        return self.browser.wait_for_element(self.task_alert, exception=False) is not None
```

**Review by jyejare — CHANGES_REQUESTED** [review]

**Review by jyejare — APPROVED** [review]

**Review by latran — APPROVED** [review]
Selection issue has been resolved.  Giving my ACK

**Review by omkarkhatavkar — APPROVED** [review]
ACK

**Pull request merged**